lms/djangoapps/grades/new/course_grade.py

```diff
@@ -123,26 +120,40 @@ def compute_and_update(self, read_only=False):
 
         If read_only is True, doesn't save any updates to the grades.
         """
-        self._log_event(log.warning, u"compute_and_update, read_only: {}".format(read_only))
         subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
         for chapter_key in self.course_structure.get_children(self.course.location):
             chapter = self.course_structure[chapter_key]
             chapter_subsection_grades = []
             for subsection_key in self.course_structure.get_children(chapter_key):
-                chapter_subsection_grades.append(
-                    subsection_grade_factory.create(self.course_structure[subsection_key], read_only=True)
+                subsection_grade = subsection_grade_factory.create(
+                    self.course_structure[subsection_key],
+                    read_only=True
                 )
+                chapter_subsection_grades.append(subsection_grade)
```

Comment:
I'll change this create-then-append logic back to how it was before, now that we don't need `subsection_grade` on its own.